Mage.Sets/src/mage/cards/c/CurseOfBounty.java

```diff
@@ -0,0 +1,154 @@
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.cards.c;
+
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.AttachEffect;
+import mage.abilities.keyword.EnchantAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.GameEvent.EventType;
+import mage.game.permanent.Permanent;
+import mage.target.TargetPlayer;
+import mage.target.targetpointer.FixedTarget;
+import java.util.UUID;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.UntapAllControllerEffect;
+import mage.filter.common.FilterNonlandPermanent;
+import mage.players.Player;
+
+/**
+ *
+ * @author Saga
+ */
+public class CurseOfBounty extends CardImpl {
+
+    public CurseOfBounty(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.ENCHANTMENT},"{1}{G}");
+        this.subtype.add(SubType.AURA, SubType.CURSE);
+
+        // Enchant player
+        TargetPlayer auraTarget = new TargetPlayer();
+        this.getSpellAbility().addTarget(auraTarget);
+        this.getSpellAbility().addEffect(new AttachEffect(Outcome.Detriment));
+        this.addAbility(new EnchantAbility(auraTarget.getTargetName()));
+
+        // Whenever enchanted player is attacked, untap all nonland permanents you control. 
+        // Each opponent attacking that player untaps all nonland permanents he or she controls.
+        Ability ability = new CurseOfBountyTriggeredAbility();
+        ability.addEffect(new UntapAllControllerEffect(new FilterNonlandPermanent()));
+        this.addAbility(ability);
+    }
+
+    public CurseOfBounty(final CurseOfBounty card) {
+        super(card);
+    }
+
+    @Override
+    public CurseOfBounty copy() {
+        return new CurseOfBounty(this);
+    }
+}
+
+class CurseOfBountyTriggeredAbility extends TriggeredAbilityImpl {
+
+    public CurseOfBountyTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new UntapAllNonlandsTargetEffect(), false);
+    }
+
+    public CurseOfBountyTriggeredAbility(final CurseOfBountyTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == EventType.DECLARED_ATTACKERS;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Permanent enchantment = game.getPermanent(this.getSourceId());
```

Comment:
So I've been looking at older curses and none of them seem to use that method. Mind explaining why this is necessary?
Also, I'm not really sure what that method name means, what's LKI stand for? Would I just replace the getPermanent method here or do I need an if clause to check for destruction before resolution and if so, how do you do that?